docs/preparing_recipes_for_conan2.md

```diff
@@ -0,0 +1,368 @@
+  new one, we should have added one more line setting the ``cmake_file_name`` value.
+
+```python
+class SomePkgConan(ConanFile):
+    name = "somepkg"
+    ...
+    def package_info(self):
+        # Legacy generators fallback the filenames for the .cmake files
+        # in the .names attribute value and generate 
+        self.cpp_info.names["cmake_find_package"] = "some-pkg" # generates module file Findsome-pkg.cmake
+        self.cpp_info.names["cmake_find_package_multi"] = "some-pkg" # generates config file some-pkg-config.cmake
+
+        self.cpp_info.set_property("cmake_target_name", "some-pkg::some-pkg") 
+        self.cpp_info.set_property("cmake_file_name", "some-pkg") # generates config file some-pkg-config.cmake
+        ...
+```
+
+Please note that if we hadn't set the ``cmake_file_name`` property, the ``CMakeDeps``
+generator would have taken the package name to generate the filename for the config file
+and the generated filename would have resulted ``somepkg-config.cmake`` instead of
+``some-pkg-config.cmake``.
+
+* Some recipes in Conan Center Index define different ``.names`` values for ``cmake_find_package``
+  and ``cmake_find_package_multi``. For these cases, besides ``cmake_target_name`` you should also set
+  the ``cmake_module_target_name`` and ``cmake_find_mode`` properties. Let's see an example:
+
+```python
+class ExpatConan(ConanFile):
+    name = "expat"
+    ...
+    def package_info(self):
+        # creates EXPAT::EXPAT target for module files FindEXPAT.cmake
+        self.cpp_info.names["cmake_find_package"] = "EXPAT"
+        # creates expat::expat target for config files expat-config.cmake
+        self.cpp_info.names["cmake_find_package_multi"] = "expat"
+        ...
+```
+
+Should translate to the code above. Please note we have added the ``cmake_find_mode``
+property for the
+[CMakeDeps](https://docs.conan.io/en/latest/reference/conanfile/tools/cmake/cmakedeps.html#properties)
+generator with value ``both``.
+
+```python
+class ExpatConan(ConanFile):
+    name = "expat"
+    ...
+    def package_info(self):
+        self.cpp_info.names["cmake_find_package"] = "EXPAT"
+        self.cpp_info.names["cmake_find_package_multi"] = "expat"
+
+        # creates EXPAT::EXPAT target for module files FindEXPAT.cmake
+        self.cpp_info.set_property("cmake_target_name", "EXPAT::EXPAT")
+        # creates expat::expat target for config files expat-config.cmake 
+        self.cpp_info.set_property("cmake_module_target_name", "expat::expat") 
+
+        # generates module file FindEXPAT.cmake
+        self.cpp_info.set_property("cmake_file_name", "EXPAT") 
+        # generates config file expat-config.cmake
+        self.cpp_info.set_property("cmake_module_file_name", "expat") 
+
+        # config is the default for CMakeDeps
+        # we set cmake_find_mode to both to generate both module and config files
+        self.cpp_info.set_property("cmake_find_mode", "both") 
```

Comment:
(Unrelated question) Can this be defined by the consumer?

Reply:
It can only be defined in recipes, maybe we should consider making something to force this from the consumers.

Reply:
Does it mean that conan won't anymore force `CMAKE_FIND_PACKAGE_PREFER_CONFIG ON` in CMakeToolchain?

Reply:
And question: what happen when we want a recipe to create both Find & config, even if they have the same names.
So we set `cmake_find_mode` to `both`, but then do we have to set `cmake_module_target_name` & `cmake_module_file_name` or do they inherit from `cmake_target_name` & `cmake_file_name`?

Reply:
Yes, if you set `cmake_find_mode` to `both`, the modules will inherit `cmake_target_name` & `cmake_file_name` so there's no need to define the `_module_` properties if the names are not different. The case of different names for find and config was just an example of an obvious case where you must set `cmake_module_target_name` but there will be more as the ones you say. I'll add a note to clarify that.

Reply:
> Does it mean that conan won't anymore force CMAKE_FIND_PACKAGE_PREFER_CONFIG ON in CMakeToolchain? Because now there is a feature to create Find file if needed, I woud say that CMAKE_FIND_PACKAGE_PREFER_CONFIG is not required anymore. It can be even misleading.

Not necessarily, because a side effect of no setting `CMAKE_FIND_PACKAGE_PREFER_CONFIG` is that CMake finds first the modules from the cmake installation directory before finding the "configs" from Conan if no "CONFIG" is provided as an argument to `find_package(openssl)`. OpenSSL is a good example. Try to consume it without `CMAKE_FIND_PACKAGE_PREFER_CONFIG` nor `find_package(openssl CONFIG)` and you would end requiring the package you don't want.

Reply:
But there is `cmake_find_mode` now, so that Find files are generated, and it should be added to all recipes with built-in CMake Find files. Therefore, modules from the cmake installation directory won't be found first.

Reply:
Yes, that is true. My concern is also removing that variable until we have a nice amount of recipes with the correct property declaration. I mean, given that the default of CMakeDeps is to generate config files (and actually always generate them) I don't see it as unnecessary, actually, I don't like the CMake default to the modules to be honest, because the conan reality is that you'll always have config files. So in that sense, it makes sense to manipulate the CMake default behavior.